code/control/DNRoot.php

```diff
@@ -630,6 +632,33 @@ public function project(SS_HTTPRequest $request) {
 	}
 
 	/**
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function starproject(SS_HTTPRequest $request) {
+		$project = $this->getCurrentProject();
+		if(!$project) {
+			return $this->project404Response();
+		}
+
+		$member = Member::currentUser();
+		if($member === null) {
+			return $this->project404Response();
+		}
+		$favProject = $member->StarredProjects()
+			->filter('DNProjectID', $project->ID)
+			->first();
+
+		if($favProject) {
```

Comment:
It may be worth noting that this is actually a "togglestarproject", it doesn't just enable the star?